app/controllers/Assessments.scala

```diff
@@ -41,17 +41,19 @@ class Assessments @Inject()(propertyLinks: PropertyLinkConnector, authenticated:
                             businessRatesAuthorisation: BusinessRatesAuthorisation)
                            (implicit val messagesApi: MessagesApi, val config: ApplicationConfig) extends PropertyLinkingController {
 
-  def assessments(authorisationId: Long) = authenticated.toViewAssessmentsFor(authorisationId) { implicit request =>
+  def assessments(authorisationId: Long, submissionId: String) = authenticated.toViewAssessmentsFor(authorisationId) { implicit request =>
     val refererOpt = request.headers.get("Referer")
 
-      propertyLinks.getLink(authorisationId) flatMap {
-        case Some(PropertyLink(_, _, _, _, _, _, _, _, Seq(), _)) => notFound
+    val pLink = if(request.organisationAccount.isAgent) propertyLinks.getClientAssessmentsWithCapacity(submissionId, authorisationId) else propertyLinks.getOwnerAssessmentsWithCapacity(submissionId, authorisationId)
```

Comment:
just because the account is an agent does not mean they are acting as an agent.

Reply:
flag now passed